subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -217,12 +217,12 @@ private boolean isNotFileSystemLoopException(@Nullable IOException e) {
         private void addFileSnapshot(Path file, String name, BasicFileAttributes attrs) {
             Preconditions.checkNotNull(attrs, "Unauthorized access to %", file);
             HashCode hash = hasher.hash(file.toFile(), attrs.size(), attrs.lastModifiedTime().toMillis());
-            RegularFileSnapshot fileSnapshot = new RegularFileSnapshot(internedAbsolutePath(file), name, hash, FileMetadata.from(attrs));
+            RegularFileSnapshot fileSnapshot = new RegularFileSnapshot(internedToString(file), name, hash, FileMetadata.from(attrs));
             builder.visit(fileSnapshot);
         }
 
-        private String internedAbsolutePath(Path file) {
-            return stringInterner.intern(file.toString());
+        private String internedToString(Object target) {
+            return stringInterner.intern(target.toString());
         }
```

Comment:
I think this doesn't make the code more understandable. Maybe inline `internedAbsolutePath()` instead? Or rename it to `internedPath(Path path)` and introduce `internedName(String name)`?

My problem with the change here is `ToString` is not well-defined at the call-site.

Reply:
Done

Reply:
it simplifies call site slightly, since there it's just noise:
```java
stringInterner.intern(file.getFileName().toString());
```
vs
```java
intern(file.getFileName().toString());
```